lib/actions/push.ts

```diff
@@ -98,30 +113,53 @@ async function getAppOwner(sdk: BalenaSDK, appName: string) {
 	return selected.extra;
 }
 
-export const push: CommandDefinition<
-	{
-		applicationOrDevice: string;
-	},
-	{
-		source: string;
-		emulated: boolean;
-		nocache: boolean;
+function parseRegistrySecrets(secretsFilename: string): RegistrySecrets {
+	let registrySecrets: RegistrySecrets = {};
+	if (secretsFilename) {
+		let isYaml = false;
+		if (/.+\.ya?ml$/i.test(secretsFilename)) isYaml = true;
+		else if (!/.+\.json$/i.test(secretsFilename))
+			throw new Error(
+				'The registry-secrets filename must end with .json, .yml or .yaml',
+			);
+
+		const raw = require('fs')
+			.readFileSync(secretsFilename)
+			.toString();
+		registrySecrets = new RegistrySecretValidator().validateRegistrySecrets(
+			isYaml ? require('js-yaml').safeLoad(raw) : JSON.parse(raw),
```

Comment:
Both `safeLoad` and `parse` can potentially throw. I'm guessing if this is the case we're just letting it bubble up to the top and stop the command, as I can't see any `try/catch` block where this is called from (assuming `await` and not Promises). Same applies for other instances of these methods.

Reply:
Yep, this is usually the reason I catch stuff and then 'translate' it into something that's obvious. :D